shell.nix

```diff
@@ -1,6 +1,5 @@
+{ pkgs ? import ./nix }:
 let
-  pkgs = import ./nix;
-
   STATE_DIR = "${builtins.toString ./.}/state";
```

Comment:
Are you sure these still work as intended?
If I recall correctly `./.` is evaluated to a store path (i.e. where the flake is in the nix store) rather than the location in the currently checked out source tree.

I want these directories to be mutable from the development environment.